src/backend/resourcemanager/resourcepool.c

```diff
@@ -893,6 +893,28 @@ int addHAWQSegWithSegStat(SegStat segstat, bool *capstatchanged)
 	else {
 		segresource = getSegResource(segid);
 		Assert(segresource != NULL);
+		uint8_t oldStatus = segresource->Stat->FTSAvailable;
+		bool statusChanged = oldStatus != segstat->FTSAvailable;
+
+		/* Check if RM process is restarted in this segment */
+		if (segresource->Stat->RMStartTimestamp != segstat->RMStartTimestamp)
```

Comment:
A comment here to illustrate the idea would be better. It is also good to mention the risk that below three condition hold:
1. rmseg reset
2. system time reset on segment node
3. segresource->Stat->RMStartTimestamp == segstat->RMStartTimestamp